### PR TITLE
Lint.hardcode style index numbers

### PR DESCRIPTION
**Pull request by wxtim**
Partially addresses #5052 - 

- Hardcodes the numbers for the Style fails
- Adds the ability to ignore individual exception codes using `--ignore` or `-n`.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [ ] If this is a bug fix, PRs raised to both master and the relevant maintenance branch.

### CODE REVIEW

**Review by MetRonnie — APPROVED** [review]

File: cylc/flow/scripts/lint.py [hunk 1]
```diff
@@ -435,6 +450,16 @@ def get_option_parser() -> COP:
...
     )
+    parser.add_option(
+        '--ignore', '-n',
+        help=(
+            'Ignore this check number.'
+        ),
+        action='append',
+        default=[],
+        dest='ignores',
+        choices=tuple([f'S{i["index"]:03d}' for i in STYLE_CHECKS.values()])
```
Comment (MetRonnie) [inline cylc/flow/scripts/lint.py:461]:
```suggestion
        metavar="CODE",
        choices=tuple([f'S{i["index"]:03d}' for i in STYLE_CHECKS.values()])
```

Comment (wxtim) [reply cylc/flow/scripts/lint.py:461]: What does this do?

Comment (MetRonnie) [reply cylc/flow/scripts/lint.py:461]: makes it display as
```
-n CODE, --ignore=CODE
```
instead of
```
-n IGNORES, --ignore=IGNORES
```

**Review by datamel — APPROVED** [review]
Code read, checked out and manually tested removing valid and non valid code. Works well with a sensible error for non-valid code. 
Thanks @wxtim.

**Comment by datamel** [comment]
Artifact upload failure is unrelated. Merging with two approvals.

**Pull request merged**